src/ranged.cpp

```diff
@@ -549,7 +549,7 @@ int player::fire_gun( const tripoint &target, int shots, item& gun )
             break;
         }
 
-        double dispersion = get_weapon_dispersion( &gun, true );
+        double dispersion = rng_normal( get_weapon_dispersion( gun ) ) + rng_normal( recoil_current() );
```

Comment:
At a minimum, we need the data Coolthulu is asking for, if I'm wrong and they're very close to each other, great, but I strongly doubt it.
Alternately, you can extract the random calls to one place, maintaining the separate sources of dispersion until it is applied.